xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+/*
+ *      Copyright (C) 2016 Team KODI
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ *------------------------------------------------------------------------------
+ * This header is only be used for Kodi itself and internally (not for add-on
+ * development) to identify the several instance types.
+ *
+ * With this reason is also no doxygen part with "///" here used.
+ * -----------------------------------------------------------------------------
+ */
+
+/*
+ * Versions of all add-on instances are defined below.
+ *
+ * This is added here and not in related header to prevent not
+ * needed includes during compile. Also have it here a better
+ * overview.
+ */
+
+#define INSTANCE_VERSION_MAIN                   "1.0.0"
```

Comment:
This was mainly to prevent dependencies!

In the coming system becomes a add-on type opened from the base add-on system, this defines there is the only part who is needed from them.

Also if a add-on want to use a inline function from there, prevent this not needed includes to add-on type instances who never used.

To this `GetInstanceVersions` function are them needed.